pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1131,57 +1109,66 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
 	} else if c.clusterFlavor == cnstypes.CnsClusterFlavorWorkload {
 		// Check if it is WCP defined feature state.
 		if _, exists := common.WCPFeatureStates[featureName]; exists {
-			log.Infof("Feature %q is a WCP defined feature state. Reading the %q configmap in %q namespace.",
-				featureName, common.WCPCapabilityConfigMapName, common.KubeSystemNamespace)
-			// Check the `wcp-cluster-capabilities` configmap in supervisor for the FSS value.
-			if wcpCapabilityFssMap == nil {
-				wcpCapabilityConfigMap, err := c.k8sClient.CoreV1().ConfigMaps(common.KubeSystemNamespace).Get(ctx,
-					common.WCPCapabilityConfigMapName, metav1.GetOptions{})
+			log.Infof("Feature %q is a WCP defined feature state. Reading the capabilities CR %q.",
+				featureName, common.WCPCapabilitiesCRName)
+
+			if len(wcpCapabilitiesMap) == 0 {
+				// TODO: remove
+				log.Infof("wcpCpabilitiesMap is empty")
+				restConfig, err := clientconfig.GetConfig()
 				if err != nil {
-					log.Errorf("failed to fetch WCP FSS configmap %q/%q. Setting the feature state "+
-						"to false. Error: %+v", common.KubeSystemNamespace, common.WCPCapabilityConfigMapName, err)
+					log.Errorf("failed to get Kubernetes config. Err: %+v", err)
 					return false
 				}
-				wcpCapabilityFssMap = wcpCapabilityConfigMap.Data
-				log.Infof("WCP cluster capabilities map - %+v", wcpCapabilityFssMap)
-			}
-			if fssVal, exists := wcpCapabilityFssMap[featureName]; exists {
-				supervisorFeatureState, err = strconv.ParseBool(fssVal)
+				cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, wcpcapapis.GroupName)
+				if err != nil {
+					log.Errorf("failed to create CnsOperator client. Err: %+v", err)
+					return false
+				}
+				wcpCapabilitiesMap, err = GetWcpCapabilitiesMap(ctx, cnsOperatorClient)
 				if err != nil {
-					log.Errorf("Error while converting %q feature state with value: %q in "+
-						"%q/%q configmap to boolean. Setting the feature state to false. Error: %+v", featureName,
-						fssVal, common.KubeSystemNamespace, common.WCPCapabilityConfigMapName, err)
+					log.Errorf("failed to get WCP capabilities map, Err: %+v", err)
 					return false
 				}
-				log.Debugf("Supervisor feature state %q in WCP cluster capabilities is set to %t", featureName,
-					supervisorFeatureState)
+				// TODO: debug?
```

Comment:
Is this ready for review? I see this todo in several places.

Reply:
I have kept few messages at INFO level for testing, removed TODOs now and changed relevant messages to DEBUG level.